receiver/faroreceiver/handler.go

```diff
@@ -0,0 +1,158 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faroreceiver
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logfmt/logfmt"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type Handler struct {
+	warnOnce sync.Once
+
+	logConsumer   consumer.Logs
+	traceConsumer consumer.Traces
+
+	logLogger   *zap.Logger
+	traceLogger *zap.Logger
+}
+
+var _ http.Handler = (*Handler)(nil)
+
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	h.warnOnce.Do(func() {
+		if h.logConsumer == nil {
+			h.traceLogger.Warn("faroreceiver logs will not be consumed!")
```

Comment:
The logger context has the receiver name already ("faro", "faro/2", ...), but it would be good to be clear why this is happening. Something like:

```suggestion
			h.traceLogger.Warn("this receiver wasn't added to a logs pipeline, logs will not be consumed")
```